device-connectors/src/testflinger_device_connectors/fw_devices/base.py

```diff
@@ -33,12 +38,47 @@ def downgrade(self):
     def check_results(self):
         raise NotImplementedError("Please, implement the check_results method")
 
-    @abstractmethod
-    def check_connectable(self):
-        raise NotImplementedError(
-            "Please, implement the check_connectable method"
-        )
-
     @abstractmethod
     def reboot(self):
         raise NotImplementedError("Please, implement the reboot method")
+
+    def check_connectable(self, timeout: int):
+        """
+        After DUT reboot, check if SSH to DUT works within a given timeout
+        period
+
+        :param timeout: wait time for regaining DUT access
+        """
+        logger.info(
+            "check and wait for %ds until %s is SSHable"
+            % (timeout, self.ipaddr)
+        )
```

Comment:
You can pass variables to log as function arguments, and this is used even in the [documentation](https://docs.python.org/3/howto/logging.html#logging-variable-data)

```suggestion
        logger.info(
            "check and wait for %ds until %s is SSHable",
            timeout, self.ipaddr
        )
```